### PR TITLE
NodeDefView: fixed errors in console when creating new nodeDef

### LINKED ISSUE #764: NodeDefEdit entity: invalid prop warning
When editing nodeDef entity, console give invalid prop warning 

Screenshot 2020-05-06 at 17 09 25

Screenshot 2020-05-06 at 17 09 12

### LINKED ISSUE #765: NodeDefEdit: validation is missing
When adding a nodeDef to an entity console shows:
```
Failed prop type: The prop `validation` is marked as required in `BasicProps`
```
See video for reproducing the issue
[Screen Recording 2020-05-06 at 17.18.52.mov.zip](https://github.com/openforis/arena/files/4587674/Screen.Recording.2020-05-06.at.17.18.52.mov.zip)

### PR DESCRIPTION
**Pull request by SteRiccio**
Fixes #765
Fixes #764

### CODE REVIEW

File: webapp/commonComponents/hooks/index.js [hunk 1]
```diff
@@ -17,5 +17,5 @@ export { default as useProfilePicture } from './useProfilePicture'
 export { default as useOnIntersect } from './useOnIntersect'
 export { default as useInterval } from './useInterval'
 export { useSurvey, useSurveyId, useSurveyInfo, useSurveyCycleKey, useOnSurveyCycleUpdate } from './useSurvey'
-export { default as useNodeDef } from './useNodeDef'
+export { useNodeDef, useNodeDefValidation, useNodeDefIsDirty } from './useNodeDef'
```
Comment (minotogna) [inline webapp/commonComponents/hooks/index.js:20]: I'm thinking that these hooks aren't generic, used only in nodeDef view? or not?

Comment (minotogna) [reply webapp/commonComponents/hooks/index.js:20]: if that's the case, you can either remove them or move in the nodeDef folder

File: server/modules/surveyRdb/service/surveyRdbService.js [hunk 2]
```diff
@@ -1,7 +1,8 @@
-import * as Survey from '@core/survey/survey'
-
 import * as SurveyManager from '@server/modules/survey/manager/surveyManager'
 import * as SurveyRdbManager from '@server/modules/surveyRdb/manager/surveyRdbManager'
```
Comment (minotogna) [inline server/modules/surveyRdb/service/surveyRdbService.js:2]: I would use relative imports here as well

File: webapp/commonComponents/hooks/index.js (hunk 1 above)
Comment (SteRiccio) [reply webapp/commonComponents/hooks/index.js:20]: uhm, you are right

File: server/modules/surveyRdb/service/surveyRdbService.js (hunk 2 above)
Comment (SteRiccio) [reply server/modules/surveyRdb/service/surveyRdbService.js:2]: sure

File: webapp/commonComponents/hooks/useNodeDef.js [hunk 3]
```diff
@@ -2,7 +2,5 @@ import { useSelector } from 'react-redux'
 
 import * as NodeDefState from '@webapp/loggedin/surveyViews/nodeDef/nodeDefState'
 
-/**
- * Gets the nodeDef from the survey views node def state.
- */
+// Gets the nodeDef from the survey views node def state.
```
Comment (minotogna) [inline webapp/commonComponents/hooks/useNodeDef.js:5]: also this doesn't look like a generic hook, or is it?

Comment (SteRiccio) [reply webapp/commonComponents/hooks/useNodeDef.js:5]: uhm, right, it's used only by NodeDefView or its inner components, I'll remove it

**Pull request merged**